### PR TITLE
Update garbage-collector.md

### PR DESCRIPTION
**Pull request by Maoni0**
the description for the GCName config is incorrect and I got rid of the link to the design doc. it's unreasonable to ask folks to read a design doc in order to use a config.

### CODE REVIEW

File: docs/core/runtime-config/garbage-collector.md [hunk 1]
```diff
@@ -556,9 +556,8 @@ Example:
 
 ## Standalone GC
 
-- Specifies a path to the library containing the garbage collector that the runtime intends to load.
-- For more information, see [Standalone GC loader design](https://github.com/dotnet/runtime/blob/main/docs/design/features/standalone-gc-loading.md).
-
+- Specifies the name of a standalone GC dll that the runtime loads in place of the one it has in the main runtime dll (coreclr.dll). This dll needs to reside in the same directory as coreclr.dll
```
Comment (gewarren) [inline docs/core/runtime-config/garbage-collector.md:559]:
```suggestion
- Specifies the name of a standalone GC assembly that the runtime loads in place of the one it has in the main runtime assembly (coreclr.dll). This assembly needs to reside in the same directory as *coreclr.dll*
```

File: docs/core/runtime-config/garbage-collector.md [hunk 2]
```diff
@@ -556,9 +556,8 @@ Example:
...
+-
```
Comment (gewarren) [inline docs/core/runtime-config/garbage-collector.md:560]:
```suggestion

```

**Review by gewarren — APPROVED** [review]

**Review by kunalspathak — APPROVED** [review]

File: docs/core/runtime-config/garbage-collector.md (hunk 1 above)
Comment (Maoni0) [reply docs/core/runtime-config/garbage-collector.md:559]: I don't think we'd want to use the word assembly here since there are native dlls?

Comment (gewarren) [reply docs/core/runtime-config/garbage-collector.md:559]: Okay. In that case should it be capitalized to "DLL"?

Comment (Maoni0) [reply docs/core/runtime-config/garbage-collector.md:559]: it can be. I'm fine either way.

Comment (gewarren) [reply docs/core/runtime-config/garbage-collector.md:559]: Okay, I think it's more correct to be capitalized so I updated my suggestion.

**Comment by Maoni0** [comment]
thanks!

**Pull request merged**